app/src/main/java/com/tomaszpolanski/androidsandbox/ViewModelBase.kt

```diff
@@ -0,0 +1,25 @@
+package com.tomaszpolanski.androidsandbox
+
+import android.arch.lifecycle.ViewModel
+import android.os.Parcelable
+import io.reactivex.Flowable
+import io.reactivex.Single
+import io.reactivex.processors.PublishProcessor
+
+abstract class ViewModelBase<S : Parcelable, in E>(
+    private val schedulerProvider: SchedulerProvider
+                                                  ) : ViewModel() {
+
+    private val intentStream = PublishProcessor.create<Pair<E, S?>>()
+
+    fun getState(restored: S?, initial: S): Flowable<S> = intentStream
```

Comment:
How do you handle changes that aren't derived directly from UI events (e.g. a data change from a notification)?

Reply:
I was thinking about having merged stream but this is still in progress